lldb/source/Plugins/Language/Swift/SwiftUnsafeTypes.cpp

```diff
@@ -0,0 +1,250 @@
+
+  static ConstString g_position("_position");
+  ValueObjectSP position_value_sp(
+      m_valobj.GetChildMemberWithName(g_position, true));
+  if (!position_value_sp || !position_value_sp->GetNumChildren())
+    return false;
+
+  ValueObjectSP some_value_sp = position_value_sp->GetChildAtIndex(0, true);
+  if (!some_value_sp || !some_value_sp->GetNumChildren())
+    return false;
+
+  CompilerType argument_type;
+
+  if (CompilerType type = some_value_sp->GetCompilerType())
+    argument_type = SwiftASTContext::GetGenericArgumentType(type, 0);
+
+  if (!argument_type.IsValid())
+    return nullptr;
+
+  m_elem_type = argument_type;
+
+  ValueObjectSP pointer_value_sp = some_value_sp->GetChildAtIndex(0, true);
+  if (!pointer_value_sp)
+    return false;
+
+  m_start_addr = pointer_value_sp->GetValueAsUnsigned(LLDB_INVALID_ADDRESS);
+
+  return true;
+}
+
+bool lldb_private::formatters::swift::UnsafeBufferPointerSummaryProvider(
+    ValueObject &valobj, Stream &stream, const TypeSummaryOptions &options) {
+
+  SwiftUnsafeBufferPointer swift_ubp(valobj);
+
+  if (!swift_ubp.Update())
+    return false;
+
+  size_t count = swift_ubp.GetCount();
+  addr_t addr = swift_ubp.GetStartAddress();
+
+  stream.Printf("%zu %s (0x%" PRIx64 ")", count,
+                (count == 1) ? "value" : "values", addr);
+
+  return true;
+}
+
+namespace lldb_private {
+namespace formatters {
+namespace swift {
+class UnsafeBufferPointerSyntheticFrontEnd : public SyntheticChildrenFrontEnd {
+public:
+  UnsafeBufferPointerSyntheticFrontEnd(lldb::ValueObjectSP valobj_sp);
+
+  virtual size_t CalculateNumChildren();
+
+  virtual lldb::ValueObjectSP GetChildAtIndex(size_t idx);
+
+  virtual bool Update();
+
+  virtual bool MightHaveChildren();
+
+  virtual size_t GetIndexOfChildWithName(ConstString name);
+
+  virtual ~UnsafeBufferPointerSyntheticFrontEnd() = default;
+
+private:
+  ExecutionContextRef m_exe_ctx_ref;
+  uint8_t m_ptr_size;
+  lldb::ByteOrder m_order;
+
+  SwiftUnsafeBufferPointer m_unsafe_ptr;
+  size_t m_element_stride;
+  DataBufferSP m_buffer_sp;
+  std::vector<ValueObjectSP> m_children;
+};
+} // namespace swift
+} // namespace formatters
+} // namespace lldb_private
+
+lldb_private::formatters::swift::UnsafeBufferPointerSyntheticFrontEnd::
+    UnsafeBufferPointerSyntheticFrontEnd(lldb::ValueObjectSP valobj_sp)
+    : SyntheticChildrenFrontEnd(*valobj_sp.get()),
+      m_unsafe_ptr(*valobj_sp.get()) {
+
+  ProcessSP process_sp = valobj_sp->GetExecutionContextRef().GetProcessSP();
+  if (!process_sp)
+    return;
+
+  if (valobj_sp)
+    Update();
+}
+
+size_t lldb_private::formatters::swift::UnsafeBufferPointerSyntheticFrontEnd::
+    CalculateNumChildren() {
+  return m_unsafe_ptr.GetCount();
+}
+
+lldb::ValueObjectSP lldb_private::formatters::swift::
+    UnsafeBufferPointerSyntheticFrontEnd::GetChildAtIndex(size_t idx) {
+  if (idx > m_unsafe_ptr.GetCount())
+    return ValueObjectSP();
```

Comment:
This check is redundant with the ones a few lines bellow. On top of that it is also bogus as a value of `num_children` would overflow.